app/models/Application.php

```diff
@@ -87,4 +89,20 @@ public static function getUserApplicationId($id)
 
         return $application = self::where('user_id', $id)->select($fields)->first();
     }
+
+    /**
+     * Call when an app is submitted to rate as 'no' if the GPA is too low
+     */
+    public function checkGPA($id)
+    {
+      $scholarship = Scholarship::getCurrentScholarship();
+      $application = self::where('id', $id)->firstOrFail();
```

Comment:
You should be able to use `$this` to access the application that you're calling this on in the controller.